include/rocksdb/table_properties.h

```diff
@@ -133,6 +133,9 @@ class TablePropertiesCollectorFactory {
  public:
   struct Context {
     uint32_t column_family_id;
+    // The level of the SST file (i.e, table), of which the properties are being
+    // collected. -1 if unknown or N/A
+    int file_level = -1;
```

Comment:
[Note]
I decided to default initialized `file_level` mainly for the reason that in our codebase, there exists creation of Context that is not aware of this newly added `file_level` and will leave it uninitialized (e.g, like [this](https://github.com/facebook/rocksdb/blob/main/utilities/table_properties_collectors/compact_on_deletion_collector_test.cc#L27-L30)). This will cause 
ci/circleci: build-linux-clang10-clang-analyze to complain.

Reply:
Can we give a `static const in kUnknownLevel = -1` or something similar, so we can potentially use it in other places if needed.

Reply:
Good call!